streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -491,14 +494,15 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
             final Produced<K, V> produced);
 
     /**
-     * Transform each record of the input stream into zero or more records in the output stream (both key and value type
+     * Transform each record of the input stream into zero or one record in the output stream (both key and value type
```

Comment:
> You probably mean flatTransform and not flatTransformValue in your comment.

Yes.

In `transformValue` it's not allowed to use `context.forward()` (neither in `transform` nor in a `punctuation`) because it would allow to change the key violating the `transformValue` contract to not change the key.

In `transform` it's ok to call `context.forward()` though as there is no such restriction.